Venmo/src/main/java/com/braintreepayments/api/VenmoReadinessResult.kt

```diff
@@ -0,0 +1,25 @@
+package com.braintreepayments.api
+
+/**
+ * Result of the [VenmoClient.isReadyToPay] check
+ */
+sealed class VenmoReadinessResult {
+
+    /**
+     * User is ready to pay with Venmo.
+     */
+    object ReadyToPay : VenmoReadinessResult()
+
+    /**
+     * User is not ready to pay with Venmo. Call [VenmoClient.showVenmoInGooglePlayStore] to
+     * allow the user to install the Venmo app.
+     */
+    object NotReadyToPay : VenmoReadinessResult()
+
+    /**
+     * An [error] occurred when determining if the user is ready to pay with Venmo.
+     */
+    class Failure(val error: Exception) : VenmoReadinessResult()
```

Comment:
Do we prefer three result options here (explicitly ready, explicitly not ready, or error), or would it be better to combine not ready and failure? The failure is only if there's an error fetching the config so we can't determine readiness. In that case it would be ReadyToPay() or NotReadyToPay(error?) with an optional error. 

Merchants likely will handle not ready and failure in the same way. But, they do have the option if a user is not ready to pay, to call `showVenmoInCGooglePayStore`. They might not want to call that on the failure case though.

Reply:
🤔 do we think merchants would call `showVenmoInGooglePayStore` with the result or when instantiating the `GooglePayClient` if they have a failure here? If it's with the `GooglePayClient` I would expect we could handle them both in 1 (ready and failure) result as they would want to parse the result in either case. That would give them time to set `showVenmoInGooglePayStore` in either case if desired.

But this is also assuming `showVenmoInGooglePayStore` is somewhere in the Google module and not here so I may be misunderstanding.

Reply:
Oh! Lol that makes way more sense. Thanks for the explanation. 🙏